src/quicktimevideo.cpp

```diff
@@ -1094,13 +1096,12 @@ void QuickTimeVideo::timeToSampleDecoder() {
   DataBuf buf(4 + 1);
   io_->readOrThrow(buf.data(), 4);
   io_->readOrThrow(buf.data(), 4);
-  size_t noOfEntries, totalframes = 0, timeOfFrames = 0;
-  noOfEntries = buf.read_uint32(0, bigEndian);
-  size_t temp;
+  uint64_t totalframes = 0, timeOfFrames = 0;
+  const uint32_t noOfEntries = buf.read_uint32(0, bigEndian);
 
-  for (unsigned long i = 1; i <= noOfEntries; i++) {
+  for (uint32_t i = 0; i < noOfEntries; i++) {
     io_->readOrThrow(buf.data(), 4);
-    temp = buf.read_uint32(0, bigEndian);
+    const uint64_t temp = buf.read_uint32(0, bigEndian);
```

Comment:
Does it makes sense to treat the variable `temp` as `uint64_t` when we are reading here a `uint32_t`? Since we are talking about frames in a video, I doubt we will ever need to process such a large video to need `uint64_t` for the totalFrames :)

Reply:
I did that because of the multiplication two lines later:

https://github.com/Exiv2/exiv2/blob/cf15cc52e01c9c4b890048bf11896278ea49271c/src/quicktimevideo.cpp#L1106

If `temp` is a `uint64_t` then the multiplication can't overflow. (Probably not a big deal if it does overflow, but I figure we might as well avoid it.)

Reply:
Would it make sense to use here some of the `SafeXXX` operations then to throw in case of overflow?

Reply:
Unfortunately, we don't currently have a `Safe::multiply`. But I think it makes sense to replace this `+=` with a `Safe::add`, which will make this code safe. I'll add a follow-up commit.